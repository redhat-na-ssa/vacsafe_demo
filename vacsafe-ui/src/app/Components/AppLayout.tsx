import * as React from 'react';
import { useHistory } from 'react-router-dom';
import {
  Page,
  Flex,
  FlexItem,
  PageHeader,
  SkipToContent,
  Title,
  Button,
} from '@patternfly/react-core';
import logo from '@app/Assets/Logo-RedHat-B-Color-RGB.png';
import {useKeycloak} from "@react-keycloak/web";

interface IAppLayout {
  children: React.ReactNode;
}

const AppLayout: React.FunctionComponent<IAppLayout> = ({ children }) => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const history = useHistory();
  const {keycloak} = useKeycloak();
  const [isMobileView, setIsMobileView] = React.useState(true);
  const onPageResize = (props: { mobileView: boolean; windowSize: number }) => {
    setIsMobileView(props.mobileView);
  };
  //TODO: extract this for re-use
  const clearAndRedirect = () => {
    keycloak.logout({redirectUri: location.origin});
    sessionStorage.clear()
  }

  //TODO: Figure out why we can't control page layout without isNavOpen of this component
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const Header = (
    <PageHeader
      isNavOpen={false}
    />
  );
  const pageId = 'primary-app-container';
  const PageSkipToContent = (
    <SkipToContent onClick={(event) => {
      event.preventDefault();
      const primaryContentContainer = document.getElementById(pageId);
      primaryContentContainer && primaryContentContainer.focus();
    }} href={`#${pageId}`}>
      Skip to Content
    </SkipToContent>
  );

  return (
    <Page
      style={{ backgroundColor: "rgb(222, 222, 222)" }}
      mainContainerId={pageId}
      onPageResize={onPageResize}
      skipToContent={PageSkipToContent}
    >
      <Flex
        alignItems={{ default: "alignItemsCenter" }}
        direction={{ default: "column" }}
        style={{ marginTop: "1rem" }}
      >
        <FlexItem alignSelf={{ default: "alignSelfFlexEnd" }}>
          <Button
            onClick={() => history.push("/contact")}
            variant="plain"
          >
            <span style={{ color: "black" }}>HELP</span>
          </Button>
          <Button onClick={clearAndRedirect} variant="plain">
            <span style={{ color: "black" }}>LOGOUT</span>
          </Button>
        </FlexItem>
        <FlexItem>
          <Title headingLevel="h1" size="3xl">
            Red Hat VCS
          </Title>
        </FlexItem>
        <FlexItem>
          <img src={logo} onClick={() => history.push("/")} alt="RH Logo" width="100px" />
        </FlexItem>
        <FlexItem style={{ width: "95%" }}>{children}</FlexItem>
      </Flex>
    </Page>
  );
}

export { AppLayout };
