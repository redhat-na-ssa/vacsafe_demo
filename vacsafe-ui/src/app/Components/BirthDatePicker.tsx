import { 
  americanDateFormat, 
  birthDateValidator, 
  birthDateRules 
} from '@app/utils/utils';
import { 
  FormGroup, 
  DatePicker 
} from '@patternfly/react-core';
import * as React from 'react';
import { Controller } from 'react-hook-form';

const BirthDatePicker = ({control, defaultBirthDate}) => {
  return (
    <Controller
      name="dateOfBirth"
      control={control}
      defaultValue={defaultBirthDate}
      render={({ field, fieldState }) => (
        <FormGroup
          label="Date of Birth"
          isRequired
          helperTextInvalid="This field is required. Please choose your date of birth in MM-DD-YYYY format."
          fieldId="date-of-birth"
          validated={fieldState.error ? "error" : "default"}
        >
          <DatePicker
            aria-label="date-of-birth"
            value={field.value}
            dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
            dateParse={americanDateFormat}
            placeholder="MM-DD-YYYY"
            helperText="Please pick your birth date."
            onChange={field.onChange}
            validators={[birthDateValidator]}
          />
        </FormGroup>
      )}
      rules={{ required: true, validate: birthDateRules, pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/ }}
    />
  )
}

export default BirthDatePicker; 