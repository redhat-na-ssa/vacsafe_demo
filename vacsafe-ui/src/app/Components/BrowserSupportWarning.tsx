import React from 'react';
import { Alert } from '@patternfly/react-core';
import UAParser from 'ua-parser-js';
const BROWSER_SUPPORT = ['Firefox', 'Chrome', 'Mobile Safari'];
const OS_SUPPORT = ['Linux', 'Mac OS', 'Windows', 'Android', 'iOS']
const parser = new UAParser();
const result = parser.getResult();

const browserCompatibility = (result: { os: { name: string; }; browser: { name: string; }; }) => {
  if (!OS_SUPPORT.includes(result.os.name)) {
    return messageRender(result.os.name);
  }
  if (!BROWSER_SUPPORT.includes(result.browser.name)) {
    return messageRender(result.browser.name);
  }
  return null;
}

const messageRender = (value : string) => (
  <Alert isInline variant="info" title="Browser Warning">
      <p>
      We noticed you are using{' '}
      <span>
       <strong>
         {value ? value : "an unsupported browser"}
       </strong>, which we do not currently support
     </span>. You may experience issues.
      </p>
      <p>
        Recommended Browsers:{' '}
          <a 
            href="https://www.google.com/chrome/"
            rel="noreferrer"
            target="_blank"
          >
            Chrome
          </a>,{' '}
          <a
            href="https://www.mozilla.org/en-US/firefox/new/"
            rel="noreferrer"
            target="_blank"
          >
            Firefox
          </a>,{' '}
          <span>
            Safari on iOS
          </span>
      </p>
    </Alert>
)

const BrowserSupportWarning: React.FunctionComponent = () => (browserCompatibility(result));

export default BrowserSupportWarning;
