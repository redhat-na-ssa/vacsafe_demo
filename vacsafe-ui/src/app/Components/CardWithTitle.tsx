import * as React from 'react';
import { 
CardTitle,
Text, TextContent, TextVariants
} from '@patternfly/react-core';

const CardWithTitle = (props) => {
  return (
     <CardTitle>
      <Text component={TextVariants.p}>
        {props.title}
      </Text>
      <TextContent>
        <Text component={TextVariants.small}>
            {props.info}
        </Text>
      </TextContent>
     </CardTitle>
  )
}

export default CardWithTitle;