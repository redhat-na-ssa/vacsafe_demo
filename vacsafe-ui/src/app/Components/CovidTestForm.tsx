import React, { useState } from "react";
import { 
  Form, 
  FormGroup, 
  TextInput, 
  FormSelect, 
  FormSelectOption, 
  DatePicker, 
  FileUpload, 
  Button, 
  Flex,
  FlexItem
} from "@patternfly/react-core";
import { CheckCircleIcon, OutlinedPaperPlaneIcon } from "@patternfly/react-icons";
import { Controller, useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";
import { 
  dateConvert, 
  americanDateFormat,
  covidTestDateRules,
  covidTestDateValidator
} from "@app/utils/utils";

const CovidTestForm = (props) => {

  const history = useHistory();
  const { control, handleSubmit } = useForm();
  const [isTestFileRejected, setIsTestFileRejected] = useState(false);

  return (
    <Form onSubmit={handleSubmit(props.onSubmit)}>
      <FormGroup
        fieldId="employee-id"
        helperText={
          props.isHRSubmit ? 
          "Make sure this matches the ID of the person you are submitting for." : 
          "Make sure this matches your ID."
        }
        label="Person ID"
      >
        <TextInput
          aria-label="employee-id"
          isDisabled
          type="text"
          value={props.employeeId}
        />
      </FormGroup>
      <FormGroup
        fieldId="employee-full-name"
        helperText={
          props.isHRSubmit ? 
          "Make sure this matches the full name of the person you are submitting for." : 
          "Make sure this matches your full name."
        }
        label="Full Name"
      >
        <TextInput
          aria-label="employee-full-name"
          isDisabled
          type="text"
          value={props.employeeName}
        />
      </FormGroup>
      {props.isHRSubmit &&
        <FormGroup
          fieldId="employee-dob"
          helperText="Make sure this matches the birth date of the person you are submitting for."
          label="Date of Birth"
        >
          <TextInput
            aria-label="employee-dob"
            isDisabled
            type="text"
            value={dateConvert(props.employeeDOB)}
          />
        </FormGroup>
      }
      <Controller
        name="test_results"
        control={control}
        rules={{ required: true }}
        defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("test_results") || "")}
        render={({ field, fieldState }) => (
          <FormGroup
            fieldId="test-results-input"
            helperText="The result of the COVID test."
            isRequired
            validated={fieldState.error ? "error" : "default"}
            label="Test Results"
            helperTextInvalid="This field is required to proceed"
          >
            <FormSelect
              aria-label="test-results-input"
              onChange={field.onChange}
              validated={fieldState.error ? "error" : "default"}
              value={field.value}
            >
              <FormSelectOption
                isPlaceholder
                isDisabled
                label="Please Select"
              />
              <FormSelectOption label="Negative" value="Negative" />
              <FormSelectOption label="Positive" value="Positive" />
              <FormSelectOption label="Inconclusive" value="Inconclusive" />
            </FormSelect>
          </FormGroup>
        )}
      />
      <Controller
        control={control}
        defaultValue={props.isHRSubmit ? "" : (dateConvert(sessionStorage.getItem("test_date") || ""))}
        name="test_date"
        render={({ field, fieldState }) => (
          <FormGroup
            label="Test Date"
            fieldId="test-results-date"
            helperText="Please pick the date this test was administered."
            helperTextInvalid="This field is required to proceed"
            isRequired
            validated={fieldState.error ? "error" : "default"}
          >
            <DatePicker
              aria-label="test-results-date"
              dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
              dateParse={americanDateFormat}
              placeholder="MM-DD-YYYY"
              onChange={field.onChange}
              value={field.value}
              validators={[covidTestDateValidator]}
            />
          </FormGroup>
        )}
        rules={{ required: true, validate: covidTestDateRules, pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/  }}
      />
      {!props.isHRSubmit &&
        <Controller
          control={control}
          defaultValue={props.onLoad}
          name="test_image_file"
          render={({ field, fieldState }) => (
            <FormGroup
              fieldId="test-record-image"
              isRequired
              label="Provide a copy of the Test Results"
              helperText="Must be a PDF, JPG, JPEG, or PNG file no larger than 4 MB"
              helperTextInvalid="Must be a PDF, JPG, JPEG, or PNG file no larger than 4 MB"
              validated={isTestFileRejected || fieldState.error ? "error" : "default"}
            >
              <FileUpload
                hideDefaultPreview
                filename={field.value["name"]}
                id="test-record-image"
                value={field.value}
                onChange={field.onChange}
                dropzoneProps={{
                  accept: ".jpg,.jpeg,.png,.pdf",
                  maxSize: 4608000,
                  onDropRejected: () => {
                    setIsTestFileRejected(true);
                  },
                }}
                validated={isTestFileRejected ? "error" : "default"}
              />
            </FormGroup>
          )}
          rules={{ required: true }}
        />
      }
      {props.isHRSubmit &&
        <Controller
          name="alternate_email"
          defaultValue={""}
          control={control}
          render={({ field, fieldState }) => (
            <FormGroup
              id="alernative-email"
              label="Alternate Email (optional)"
              helperText="Optional email for notifications"
              helperTextInvalid="This needs to be a valid email address"
              fieldId="alernative-email"
              validated={fieldState.error ? "error" : "default"}
            >
              <TextInput
                aria-label="alernative-email"
                onChange={field.onChange}
                placeholder="example@mail.com"
                type="email"
                value={field.value}
              />
            </FormGroup>
          )}
          rules={{
            required: false,
            pattern: /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}/g,
          }}
        />
      }
      {props.isHRSubmit ? 
        <FormGroup fieldId="accept-test-buttons">
          <Flex>
            <FlexItem align={{ default: "alignRight" }}>
              <Button
                icon={<OutlinedPaperPlaneIcon />}
                variant="primary"
                type="submit"
                isDisabled={props.isFormSubmitting}
                isLoading={props.isFormSubmitting}
              >
                Submit
              </Button>
            </FlexItem>
          </Flex>
        </FormGroup>
        :
        <FormGroup fieldId="accept-test-buttons">
          <Button 
            icon={<CheckCircleIcon />} 
            variant="primary" 
            type="submit"
          >
            Accept
          </Button>
          <Button 
            variant="plain" 
            onClick={() => history.push("/attestmenu")}
          >
            Back
          </Button>
        </FormGroup>
      }
    </Form>
  )
};

export default CovidTestForm;