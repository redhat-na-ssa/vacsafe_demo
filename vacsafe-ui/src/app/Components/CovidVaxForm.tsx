import React, { useState } from "react";
import {
  Button,
  DatePicker,
  FileUpload,
  Flex,
  FlexItem,
  Form,
  FormGroup,
  FormSelect,
  FormSelectOption,
  TextInput,
} from "@patternfly/react-core";
import { CheckCircleIcon, OutlinedPaperPlaneIcon } from "@patternfly/react-icons";
import { useHistory } from "react-router-dom";
import { useForm, Controller } from "react-hook-form";
import { 
  dateConvert, 
  americanDateFormat, 
  covidVaxOneDateRules, 
  covidVaxOneDateValidator,
  covidVaxTwoDateRules,
  covidVaxTwoDateValidator
} from "@app/utils/utils";

const vendorOptions = [
  { value: "AstraZeneca", label: "AstraZeneca" },
  { value: "Janssen", label: "Janssen" },
  { value: "Johnson", label: "Johnson & Johnson" },
  { value: "Moderna", label: "Moderna" },
  { value: "Novavax", label: "Novavax" },
  { value: "Pfizer", label: "Pfizer" },
];

const CovidVaxForm = (props) => {
  const history = useHistory();
  const { control, handleSubmit, getValues } = useForm();
  const [isRejected, setIsRejected] = useState(false);

  return (
    <Form onSubmit={handleSubmit(props.onSubmit)}>
      <FormGroup 
        fieldId="employee-id" 
        helperText={
          props.isHRSubmit ? 
          "Make sure this matches the ID of the person you are submitting for." : 
          "Make sure this matches your ID."
        }
        label="Person ID"
      >
        <TextInput
          aria-label="employee-id"
          isDisabled
          type="text"
          value={props.employeeId}
        />
      </FormGroup>
      <FormGroup 
        fieldId="employee-full-name" 
        helperText={
          props.isHRSubmit ? 
          "Make sure this matches the full name of the person you are submitting for." : 
          "Make sure this matches your full name."
        }
        label="Full Name"
      >
        <TextInput
          aria-label="employee-full-name"
          isDisabled
          type="text"
          value={props.employeeName}
        />
      </FormGroup>
      {props.isHRSubmit &&
        <FormGroup
          fieldId="employee-dob"
          helperText="Make sure this matches the birth date of the person you are submitting for."
          label="Date of Birth"
        >
          <TextInput
            aria-label="employee-dob"
            isDisabled
            type="text"
            value={dateConvert(props.employeeDOB)}
          />
        </FormGroup>
      }
      <Controller
        name="vaccine"
        control={control}
        defaultValue={props.isHRSubmit ? "" : (sessionStorage.getItem("vaccine") || "")}
        render={({ field, fieldState }) => (
          <FormGroup
            helperText={
              props.isHRSubmit ? 
              "Which vaccine was administered?" : 
              "Which vaccine did you receive?"
            }
            helperTextInvalid="This field is required, please make a selection."
            isRequired
            label="Vaccine"
            fieldId="vaccine-vendor"
            validated={fieldState.error ? "error" : "default"}
          >
            <FormSelect
              aria-label="vaccine-vendor"
              value={field.value}
              onChange={field.onChange}
            >
              <FormSelectOption
                isPlaceholder
                isDisabled
                label="Please Select"
              />
              {vendorOptions.map((option, index) => (
                <FormSelectOption
                  key={index}
                  value={option.value}
                  label={option.label}
                />
              ))}
            </FormSelect>
          </FormGroup>
        )}
        rules={{ required: true }}
      />
      <Controller
        name="vax_date_1"
        control={control}
        defaultValue={props.isHRSubmit ? "" : (dateConvert(sessionStorage.getItem("vax_date_1") || ""))}
        render={({ field, fieldState }) => (
          <FormGroup
            helperText="Please choose the date of the first shot."
            helperTextInvalid="This field is required, please enter a valid date."
            fieldId="shot-1-date"
            isRequired
            label="Date of Shot 1"
            validated={fieldState.error ? "error" : "default"}
          >
            <DatePicker
              aria-label="shot-1-date"
              onChange={field.onChange}
              dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
              dateParse={americanDateFormat}
              placeholder="MM-DD-YYYY"
              value={field.value}
              validators={[covidVaxOneDateValidator]}
            />
          </FormGroup>
        )}
        rules={{ required: true, validate: covidVaxOneDateRules, pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/ }}
      />
      <Controller
        name="vax_date_2"
        control={control}
        defaultValue={props.isHRSubmit ? "" : (dateConvert(sessionStorage.getItem("vax_date_2") || ""))}
        render={({ field, fieldState }) => (
          <FormGroup
            helperText="Please choose the date of the second shot."
            helperTextInvalid="This field is required, please enter a valid date."
            fieldId="shot-2-date"
            label="Date of Shot 2 (if applicable)"
            validated={fieldState.error ? "error" : "default"}
          >
            <DatePicker
              aria-label="shot-2-date"
              onChange={field.onChange}
              dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
              dateParse={americanDateFormat}
              placeholder="MM-DD-YYYY"
              value={field.value}
              validators={[ (date) => covidVaxTwoDateValidator(date, new Date(getValues("vax_date_1").replace(/-/g,'/'))) ]}
            />
          </FormGroup>
        )}
        rules={{ validate: (data) => covidVaxTwoDateRules(data, getValues("vax_date_1")), pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/ }}
      />
      {!props.isHRSubmit &&
        <Controller
          control={control}
          defaultValue={props.onLoad}
          name="vax_image"
          render={({ field, fieldState }) => (
            <FormGroup
              fieldId="vaccination-record-image"
              isRequired
              label="Select Image of Vaccination Record"
              helperText="Copy of CDC/Vaccine Record Card, note or receipt from doctor with name, dates and vaccine type, or a printout from the CVMS system."
              helperTextInvalid="Must be a PDF, JPG, JPEG, or PNG file no larger than 4 MB"
              validated={isRejected || fieldState.error ? "error" : "default"}
            >
              <FileUpload
                hideDefaultPreview
                filename={field.value["name"]}
                id="vaccination-record-image"
                value={field.value}
                onChange={field.onChange}
                dropzoneProps={{
                  accept: ".jpg,.jpeg,.png,.pdf",
                  maxSize: 4608000,
                  onDropRejected: () => {
                    setIsRejected(true);
                  },
                }}
                validated={isRejected ? "error" : "default"}
              />
            </FormGroup>
          )}
          rules={{ required: true }}
        />
      }
      {props.isHRSubmit && 
        <Controller
          name="alternate_email"
          defaultValue={""}
          control={control}
          render={({ field, fieldState }) => (
            <FormGroup
              id="alernative-email"
              label="Alternate Email (optional)"
              helperText="Optional email for notifications"
              helperTextInvalid="This needs to be a valid email address"
              fieldId="alernative-email"
              validated={fieldState.error ? "error" : "default"}
            >
              <TextInput
                aria-label="alernative-email"
                onChange={field.onChange}
                placeholder="example@mail.com"
                type="email"
                value={field.value}
              />
            </FormGroup>
          )}
          rules={{
            required: false,
            pattern: /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}/g,
          }}
        />
      }
      {props.isHRSubmit ? 
        <FormGroup fieldId="accept-button">
          <Flex>
            <FlexItem align={{ default: "alignRight" }}>
              <Button
                icon={<OutlinedPaperPlaneIcon />}
                variant="primary"
                type="submit"
                isDisabled={props.isFormSubmitting}
                isLoading={props.isFormSubmitting}
              >
                Submit
              </Button>
            </FlexItem>
          </Flex>
        </FormGroup>
        :
        <FormGroup fieldId="accept buttons">
          <Button 
            icon={<CheckCircleIcon />} 
            type="submit" 
            variant="primary"
          >
            Accept
          </Button>
          <Button 
            onClick={() => history.push("/attestmenu")} 
            variant="plain"
          >
            Back
          </Button>
        </FormGroup>
      }
    </Form>
  )
}

export default CovidVaxForm;