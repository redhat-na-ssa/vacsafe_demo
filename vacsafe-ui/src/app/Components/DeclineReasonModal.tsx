import * as React from 'react';
import { Button, Modal, ModalVariant, Radio, Stack } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';
import { useState } from 'react';

const DeclineReasonModal = ({...props}) => {
  const [checked, setChecked] = useState(""); 
  const declineReasons = [
    "Attachments not legible",
    "Supporting document missing vaccine information",
    "Supporting document missing name and/or DOB",
    "Not a valid form of supporting documentation",
    "Supporting documentation mismatch with employee information",
    "Supporting documentation provided is a copy of the wrong side of the card"
  ]

  const handleRadioChange = (_, event) => {
    props.setRejectReason(event.currentTarget.value);
    setChecked(event.currentTarget.name);
  }

  return (
    <Modal
      actions={[
        <Button
          icon={<CheckCircleIcon />}
          isDisabled={props.isSendingReview}
          isLoading={props.isSendingReview}
          key="confirm"
          variant="primary"
          onClick={() => props.sendReview(false)}
        >
          Confirm
        </Button>,
        <Button
          key="cancel"
          onClick={() => props.setIsModalOpen(false)}
          variant="plain"
        >
          Cancel
        </Button>
      ]}
      isOpen={props.isModalOpen}
      onClose={() => props.setIsModalOpen(false)}
      title="Reason For Decline - Select Most Appropriate"
      variant={ModalVariant.medium}
    >
      <Stack hasGutter>
        {declineReasons.map((reason, index) => (
          <Radio
            key={"radio-" + index}
            isChecked={checked === "radio-" + index}
            name={"radio-" + index}
            onChange={handleRadioChange}
            label={reason}
            id={"radio-controlled-" + index}
            value={reason}
          />
        ))}
      </Stack>
    </Modal>
  )
}

export default DeclineReasonModal;