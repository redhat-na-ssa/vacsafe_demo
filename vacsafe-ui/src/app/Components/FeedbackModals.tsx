import * as React from 'react';
import { Button, Modal, ModalVariant } from '@patternfly/react-core';


const ResponseErrorModal = (props) => {
  const ServerError = () => {
    if(props.errorState === 400) {
      return <span> Form entry is incorrect. Please review form entries. </span>
    }
    else {
      return <span> Server is unable to respond. Please try again later. </span>
    }
  }

  return (
    <Modal
      variant={ModalVariant.small}
      title="Error Submitting Form"
      isOpen={props.isModalOpen}
      onClose={() => props.setIsModalOpen(false)}
      actions={[
        <Button key="cancel" variant="tertiary" onClick={() => props.setIsModalOpen(false)}>
          Close
        </Button>
      ]}
    >
      <ServerError />
    </Modal>
  )
}

const GenericResponseModal = (props) => {
  return (
    <Modal
      variant={ModalVariant.small}
      title={props.title}
      onClose={() => props.setIsModalOpen(false)}
      isOpen={props.isModalOpen}
      actions={[
        <Button key="cancel" variant="tertiary" onClick={() => props.setIsModalOpen(false)}>
          Close
        </Button>
      ]}
    >
      {props.message}
    </Modal>
  )
}

export { 
  ResponseErrorModal, 
  GenericResponseModal
}