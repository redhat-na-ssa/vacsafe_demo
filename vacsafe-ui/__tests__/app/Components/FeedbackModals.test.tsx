import React from 'react';
import { shallow } from 'enzyme';
import { GenericResponseModal, ResponseErrorModal }  from '../../../src/app/Components/FeedbackModals';

describe('<GenericResponseModal />', ()=> {
    it('should be defined', () => {
        expect(GenericResponseModal).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<GenericResponseModal />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<GenericResponseModal />)
        expect(wrapper).toMatchSnapshot()
    })
})

describe('<ResponseErrorModal />', ()=> {
    it('should be defined', () => {
        expect(ResponseErrorModal).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<ResponseErrorModal />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<ResponseErrorModal />)
        expect(wrapper).toMatchSnapshot()
    })
})
