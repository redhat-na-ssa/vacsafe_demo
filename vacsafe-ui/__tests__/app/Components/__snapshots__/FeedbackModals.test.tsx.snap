// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericResponseModal /> should match existing snapshot 1`] = `
<Modal
  actions={
    Array [
      <Button
        onClick={[Function]}
        variant="tertiary"
      >
        Close
      </Button>,
    ]
  }
  appendTo={[Function]}
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isOpen={false}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title=""
  titleIconVariant={null}
  titleLabel=""
  variant="small"
/>
`;

exports[`<ResponseErrorModal /> should match existing snapshot 1`] = `
<Modal
  actions={
    Array [
      <Button
        onClick={[Function]}
        variant="tertiary"
      >
        Close
      </Button>,
    ]
  }
  appendTo={[Function]}
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isOpen={false}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Error Submitting Form"
  titleIconVariant={null}
  titleLabel=""
  variant="small"
>
  <ServerError />
</Modal>
`;
